lte/gateway/c/core/oai/tasks/amf/amf_authentication.cpp

```diff
@@ -551,6 +555,31 @@ int amf_proc_authentication_complete(
       auth_proc->retransmission_count++;
       OAILOG_WARNING(
           LOG_NAS_AMF, "Authentication failure due to RES,XRES mismatch \n");
+      OAILOG_STREAM_HEX(
+          OAILOG_LEVEL_INFO, LOG_NAS_AMF, "Stored RAND: ",
+          (char*) (amf_ctx->_vector[auth_proc->ksi % MAX_EPS_AUTH_VECTORS]
+                       .rand),
+          AUTH_RAND_SIZE);
+      OAILOG_STREAM_HEX(
+          OAILOG_LEVEL_INFO, LOG_NAS_AMF, "Stored AUTN: ",
+          (char*) (amf_ctx->_vector[auth_proc->ksi % MAX_EPS_AUTH_VECTORS]
+                       .autn),
+          AUTH_RAND_SIZE);
+      OAILOG_STREAM_HEX(
+          OAILOG_LEVEL_INFO, LOG_NAS_AMF, "Stored XRES*: ",
+          (char*) (amf_ctx->_vector[auth_proc->ksi % MAX_EPS_AUTH_VECTORS]
+                       .xres_star),
+          AUTH_RAND_SIZE);
+      OAILOG_STREAM_HEX(
+          OAILOG_LEVEL_INFO, LOG_NAS_AMF, "Received RES*in AUTH RSP:",
+          (char*) (msg->autn_response_parameter.response_parameter),
+          AUTH_RAND_SIZE);
+      PLMN_T_TO_MCC_MNC(
+          amf_ctx->originating_tai.plmn, mcc, mnc, mnc_digit_length);
+      snprintf(
+          snni_buffer, sizeof(snni_buffer),
+          "5G:mnc%03d.mcc%03d.3gppnetwork.org", mnc, mcc);
+      OAILOG_INFO(LOG_NAS_AMF, "Serving network name: %s\n", snni_buffer);
```

Comment:
I would just suggest to try something like:
#define SNNI_FMT "5G:mnc%03d.mcc%03d.3gppnetwork.org"
char snni_buffer[sizeof(SNNI_FMT) - 1] = {};
...
snprintf(
          snni_buffer, sizeof(snni_buffer),
          SNNI_FMT, mnc, mcc);

So that if somebody wants to change the trace, he will not have to make mistakes in counting the size of formatted string to dimension the array. But your code is safe anyway.

Reply:
Done